venv/pysnes/address_computation_helper.py

```diff
@@ -0,0 +1,67 @@
+
+
+
+# some addresses are wrapped at the bank XX bank boundery (64 KB - 16 Bit addr)
+# so if you read two bytes from XXFFFF the second byte will be read
+# from XX0000 and not from X100000!
+#
+# There are mixed address modes. For example the computation of an
+# address pointer using dp register is wrapped.
+#
+# while the address itself can cross bank bounderys.
+# Direct page (dp register) and stack pointer are wrapped at the zero bank.
+# Most likely there are bugs in wrapping implementations.
+def compute_wrapped_addr(address):
+    return address & 0x00FFFF
+
+# OPCODE abs
+def abs(address, DBR):
+    return (DBR << 16) + address # no wrapping
```

Comment:
I believe that this kind of bit magic is the hardest thing to understand for someone new. Maybe explicitly explain here why DBR needs to be shifted by 16 bits here. e.g. address is 0xXXYYYY where XX is DBR.